application-analytics-ui/src/main/resources/Analytics/Code/Macros/VelocityMacros.xml

```diff
@@ -74,5 +74,122 @@
     #set ($discard = $resultList.add($element))
   #end
 #end
+
+#macro (subSet $object $start $end $resultList)
+  #set($resultList = [])
+  #set ($resultList = [])
+  #set ($end = $mathtool.sub($end,1))
+  #foreach($i in [$start..$end])
+    #set($element = $object.get($i))
+    #set($dummy = $resultList.add($element))
+  #end
+#end
+
+#macro(analyticsActions $parameter)
+&lt;div class="analyticsActions"&gt;
+  &lt;div class="analyticsRowEvolution"  data-parameter=$escapetool.url($parameter)&gt;
+    &lt;a class="action" title="$escapetool.xml($services.localization.render(
+      'analytics.action.hover.rowEvolution'))" href="#"&gt;
+      &lt;span class="action-icon"&gt;$services.icon.renderHTML('eye')&lt;/span&gt;&lt;span class="action-label"&gt;&lt;/span&gt;
+    &lt;/a&gt;
+  &lt;/div&gt;
+&lt;/div&gt;
+#end
+
+#macro(durationSelect $className $options $id)
+&lt;select i class="$className form-control" aria-labelledby=$id&gt;
+  #foreach($option in $options)
+    &lt;option value="$option"&gt;$option&lt;/option&gt;
+  #end
+&lt;/select&gt;
+#end
+
+
+#macro(durationOption $value)
+  &lt;option value="$value"&gt;
+    $services.localization.render('analytics.rowEvolution.modal.' + $value.toLowerCase() + '.label')
+  &lt;/option&gt;
+#end
+
+#macro(durationSelectList $class $options)
+  &lt;select class="$class form-control" aria-label='duration'&gt;
+    #foreach($option in $options)
+      &lt;option value="$option"&gt;$option&lt;/option&gt;
+    #end
+  &lt;/select&gt;
+#end
+
+#macro(rowEvolutionFieldSelect $matomoFields $id)
+  &lt;select id="rowEvolutionPeriod$id"  class="form-control rowEvolutionField"&gt;
+    #foreach($key in $matomoFields.keySet())
+      &lt;option value="$matomoFields.get($key)"&gt;
+        $services.localization.render('analytics.rowEvolution.field.' + $key)
+      &lt;/option&gt;
+    #end
+  &lt;/select&gt;
+#end
+
+#macro(modalRowEvolution $matomoFields $macroName $id)
```

Comment:
This should be refactored, as it is too long. Maybe you can extract the `modal-body` content in a separate macro?
Also, check that there are lines that exceed the 120ch limit